node/src/components/consensus/protocols/zug.rs

```diff
@@ -2599,3 +2620,54 @@ mod specimen_support {
         }
     }
 }
+
+mod registered_sync {
+    use crate::types::{DataSize, NodeRng};
+    use casper_types::{TimeDiff, Timestamp};
+    use rand::Rng as _;
+    use serde::{Deserialize, Serialize};
+    use std::collections::BTreeMap;
+
+    #[derive(Default, DataSize, Clone, Serialize, Deserialize, Debug, PartialEq, Eq, Hash)]
+    pub struct RegisteredSync(BTreeMap<RandomId, Timestamp>);
+
+    #[derive(
+        DataSize, Copy, Clone, Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Ord, PartialOrd,
+    )]
+    pub struct RandomId(u64);
+
+    impl RegisteredSync {
+        /// Prunes entries older than one minute.
+        fn prune_old(&mut self) {
+            const ONE_MIN: TimeDiff = TimeDiff::from_seconds(60);
+
+            self.0.retain(|_, timestamp| timestamp.elapsed() > ONE_MIN);
```

Comment:
I think this should be the reverse, shouldn't it? We want to only retain the entries that are _younger_ than one minute, not older.

Reply:
🤦🏻